node/heartbeat/heartbeatMessageInfo.go

```diff
@@ -192,6 +195,5 @@ func (hbmi *heartbeatMessageInfo) GetIsActive() bool {
 func (hbmi *heartbeatMessageInfo) GetIsValidator() bool {
 	hbmi.updateMutex.Lock()
 	defer hbmi.updateMutex.Unlock()
-	isValidator := hbmi.isValidator
-	return isValidator
+	return hbmi.peerType == string(core.EligibleList)
```

Comment:
it is a validator if it is in waiting list as well

Reply:
fixed